### PR TITLE
Add JSON data file locally

### LINKED ISSUE #150: Package cache json file locally
Add (using GitHub Actions/part of the pipeline) cache file to the repo. Store within the `data` sub-directory of the `fake-useragent` directory.

Add to `pyproject.toml` file:

```toml
[tool.setuptools.package-data]
"fake-useragent.data" = ["*.json"]
```

Using importlib to retrieve the data file:

```py
try:
  from importlib.resources import files
except ImportError:
  from importlib_resources import files

local_data = files('fake-useragent.data').joinpath('cache.json').read_text()
```

And add the data folder to [PyPi package as data files](https://setuptools.pypa.io/en/stable/userguide/datafiles.html).

### PR DESCRIPTION
**Pull request by melroy89**
Fixes #150, #139, #149, #151

Note: Data from useragentstring.com, can also not always be retrieved during GitHub Actions. We add the JSON Lines file to the package. Which solves a lot of issues at once.

- [x] Do not depend on external data sources during run-time. Embed the data file locally within the pypi package.
- [x] Remove centralized caching server from code.
- [x] Introduce option to turn off/on local data retrieval
- [x] Fallback to data source (which is currently useragentstring.com)
- [x] Fix testcases
- [x] Do no depend on the internet during testing (fully mocked/faked/patched)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -44,6 +44,7 @@ content-type = "text/markdown"
 Homepage = "https://github.com/fake-useragent/fake-useragent"
 
 [tool.setuptools]
-packages = ["fake_useragent"]
 zip-safe = false
-include-package-data = true
+
+[tool.setuptools.package-data]
+"fake_useragent.data" = ["*.json"]
```
Comment (jayvdb) [inline pyproject.toml]: eol at eof

File: pyproject.toml [hunk 2]
```diff
@@ -44,6 +44,7 @@ content-type = "text/markdown"
 Homepage = "https://github.com/fake-useragent/fake-useragent"
 
 [tool.setuptools]
-packages = ["fake_useragent"]
 zip-safe = false
-include-package-data = true
```
Comment (jayvdb) [inline pyproject.toml:49]: i think you sure this isnt needed?

File: src/fake_useragent/fake.py [hunk 3]
```diff
@@ -10,27 +10,22 @@
 class FakeUserAgent:
     def __init__(
         self,
-        cache=True,
-        use_cache_server=True,
-        path=settings.DB,
+        use_external_data=False,
+        tmp_path=settings.DB,
```
Comment (jayvdb) [inline src/fake_useragent/fake.py]: `tmp_path` doesnt feel like a good name for this.  The file provided in this arg isnt a temporary file.  It is more like a cache file...?

File: pyproject.toml (hunk 2 above)
Comment (melroy89) [reply pyproject.toml:49]: It's default true

File: src/fake_useragent/fake.py (hunk 3 above)
Comment (melroy89) [reply src/fake_useragent/fake.py]: True you are right

**Comment by melroy89** [comment]
@jayvdb Can you confirm the tests are running fine in isolation as well (ie no internet)?

**Pull request merged**